frontend/src/host_orchestrator/orchestrator/artifacts/artifacts.go

```diff
@@ -101,12 +90,12 @@ func (h *Manager) GetKernelBundle(buildID, target string, fetcher Fetcher) (stri
 		}
 		return outDir, nil
 	}
-	return h.syncDownload(buildID+target+"kernel", f)
+	return h.registry.Get(outDir).Run(f)
 }
 
 func (h *Manager) GetBootloaderBundle(buildID, target string, fetcher Fetcher) (string, error) {
+	outDir := fmt.Sprintf("%s/%s_%s__bootloader", h.rootDir, buildID, target)
 	f := func() (string, error) {
-		outDir := fmt.Sprintf("%s/%s_%s__bootloader", h.rootDir, buildID, target)
 		if err := createDir(outDir); err != nil {
```

Comment:
nit: Creating the directory is better suited for the registry than the download function.

Reply:
we've been doing it this way because `cvd fetch` needs to create the out directory to set relevant permissions.

Reply:
I don't understand what `fetch_cvd` has to do with this. My request is just to move the call to `createDir(outDir)` in line 105 to when the `downloadContext` is created around line 151 (probably in a `newDownloadContext` function). Functionality won't change, only the organization of the code will.

I just noticed that all functors but one call `createDir`: Why does GetCVDBundle not call `createDir`?

Reply:
in the GetCVDBundle, the fetcher is the one that creates the directory, which the fetch implementation we use right now, ends up executing `fetch_cvd`. This is an implementation detail that could be handled in a follow up PR as it has been like this. Leaving the responsibility to the fetcher will be the way to go.

Reply:
we can address in a follow up PR, but I don't like to leave it to the fetcher: When the registry "acquires" a directory it should expect that directory to not exist yet and use the opportunity to assert that and then create it.

Reply:
sounds good.